homeassistant/components/barry/config_flow.py

```diff
@@ -0,0 +1,88 @@
+"""Adds config flow for Barry integration."""
+# pylint: disable=attribute-defined-outside-init
+from pybarry import Barry, InvalidToken
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_ACCESS_TOKEN
+
+from .const import DOMAIN  # pylint:disable=unused-import
+
+
+class BarryConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Barry integration."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    async def async_step_import(self, import_info):
+        """Set the config entry up from yaml."""
+        return await self.async_step_user(import_info)
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        data_schema = vol.Schema({vol.Required(CONF_ACCESS_TOKEN): str})
+
+        if self._async_current_entries():
+            return self.async_abort(reason="already_configured")
+
+        if user_input is not None:
+            access_token = user_input[CONF_ACCESS_TOKEN].replace(" ", "")
+
+            barry_connection = Barry(
+                access_token=access_token,
+            )
+
+            errors = {}
+            try:
+                barry_connection.get_all_metering_points(check_token=True)
+            except InvalidToken:
+                errors[CONF_ACCESS_TOKEN] = "invalid_access_token"
```

Comment:
Best would be to catch all errors, and eventually have a broad exception returning an `unknown_error`.

Reply:
Thanks for the suggestion. Will replace `InvalidToken` exception with `Exception` to cover all the errors and return `unknown_error`.

Reply:
You don't need to replace this exception, best would to catch InvalidToken and add a broad except all for other expections.

Reply:
I tried to add Broad exception, but the pipeline is failing due to that broad exception. Can you please help me out here?

Reply:
Have a look at the Shelly integration for example; https://github.com/home-assistant/core/blob/dev/homeassistant/components/shelly/config_flow.py#L93.

You need to disable pylint for this line, since it is allowed for this usecase.